src/sidebar/FeedList/Folder.tsx

```diff
@@ -94,17 +124,26 @@ const Folder = (props: Props) => {
             return;
         }
 
-        // TODO determine drop position (top, center, bottom) based on drop target bounding box and drag position
-        // use that information (local state) to highlight (line, highlight label) and use it fro drop effect (before, insert, after)
         if (!props.disabled) {
-            setDraggedOver(true);
+            if (dragged?.nodeType === NodeType.Feed && props.nodeType === NodeType.Folder) {
+                // feeds can only be inserted into folders
+                setRelativDropPosition(RelativeDragDropPosition.Middle);
+            } else if (dragged?.nodeType === NodeType.Feed && props.nodeType === NodeType.Feed) {
+                // feeds can only be sorted
+                setRelativDropPosition(relativeDragDropPosition(event, 0.5));
+            } else if (dragged?.nodeType === NodeType.Folder && props.nodeType === NodeType.Folder) {
+                setRelativDropPosition(relativeDragDropPosition(event, 0.15));
+            } else {
+                throw new Error(`${dragged?.nodeType} can not be dropped on ${props.nodeType}.`);
```

Comment:
dragged can be undefined if the drop happens to fast and the store is not yet updated

Reply:
https://github.com/dermeck/feeds-sidebar/pull/157/commits/b8e5f5b95ea2a61c56373a4ff533f903e54f7a35